iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergOutputCommitter.java

```diff
@@ -307,27 +307,24 @@ public void commitJobs(List<JobContext> originalContextList, Operation operation
     }
   }
 
-  private List<OutputTable> collectOutputs(List<JobContext> jobContextList) {
-    return jobContextList.stream()
-      .flatMap(jobContext -> HiveIcebergStorageHandler.outputTables(jobContext.getJobConf()).stream()
-        .map(output -> new OutputTable(
-          HiveIcebergStorageHandler.catalogName(jobContext.getJobConf(), output),
-          output,
-          SessionStateUtil.getResource(jobContext.getJobConf(), output)
+  private Multimap<OutputTable, JobContext> collectOutputs(List<JobContext> jobContextList) {
+    Multimap<OutputTable, JobContext> outputs =
```

Comment:
Why MultiMap? Till now we only considered one JobContext, will this generate multiple JobContexts per OutputTable.

Reply:
We have multiple JobContext per OutputTable, nothing changed here, I just used a more convenient collection. Update and Merge operations are represented as multi-insert creating multiple contexts